mkosi/distributions/gentoo.py

```diff
@@ -268,9 +233,8 @@ def fetch_fix_stage3(self, state: MkosiState, root: Path) -> None:
         # and more... so we can gladly escape all this hideousness!
         ###########################################################
         with urllib.request.urlopen(stage3tsf_path_url) as r:
-            args_profile = "nomultilib"
-            # 20210711T170538Z/stage3-amd64-nomultilib-systemd-20210711T170538Z.tar.xz 214470580
-            regexp = f"^[0-9TZ]+/stage3-{self.arch}-{args_profile}-systemd-[0-9TZ]+[.]tar[.]xz"
+            # e.g.: stage3-amd64-nomultilib-systemd-mergedusr-20221127T170156Z.tar
+            regexp = f"^[0-9TZ]+[/]stage3[-]{self.arch}[-]nomultilib[-]systemd[-]mergedusr[-[0-9TZ]+[.]tar[.]xz"
```

Comment:
The parentheses are not balanced (why escape the dashes anyway?) and the regex doesn't match the example above, since the initial `[0-9TZ]+/` is and the `.xz` at the end are missing, how about
```suggestion
            # e.g.: 20221127T170156Z/stage3-amd64-nomultilib-systemd-mergedusr-20221127T170156Z.tar.xz
            regexp = f"^[0-9]+T[0-9]+Z/stage3-{arch}-nomultilib-systemd-mergedusr-[0-9]+T[0-9]+Z\.tar\.xz"
```

Reply:
do we need the `[0-9TZ]+/` at the beginning and the `.xz` at the end?

Reply:
You tell me :) My point is mainly that the example and the regexp should actually match and document what is expected or needed. Also, the groups for dashes and dots are still a bit weird and the square brackets don't match (the one after `mergedusr` never closes).

Reply:
you're right. fixed with your suggestion except for `\.`, i get `invalid escape sequence` here for some reason. python version? editor stuff?

Reply:
That's with your editor then, I assume, I ran this directly in the REPL (3.9 and 3.10)

Reply:
changed to `\.`